cluster-autoscaler/core/utils.go

```diff
@@ -121,20 +122,24 @@ func FilterOutSchedulable(unschedulableCandidates []*apiv1.Pod, nodes []*apiv1.N
 	loggingQuota := glogx.PodsLoggingQuota()
 
 	for _, pod := range unschedulableCandidates {
-		if schedulable, found := podSchedulable.get(pod); found {
-			if !schedulable {
-				unschedulablePods = append(unschedulablePods, pod)
-			} else {
-				glogx.V(4).UpTo(loggingQuota).Infof("Pod %s marked as unschedulable can be scheduled (based on simulation run for other pod owned by the same controller). Ignoring in scale up.", pod.Name)
+		schedulable, _, found := podSchedulable.get(pod)
+		if found && schedulable {
+			glogx.V(4).UpTo(loggingQuota).Infof("Pod %s marked as unschedulable can be scheduled (based on simulation run for other pod owned by the same controller). Ignoring in scale up.", pod.Name)
+		}
+
+		if !found {
+			nodeName, err := predicateChecker.FitsAny(pod, nodeNameToNodeInfo)
+			schedulable = err == nil
+			// err returned from FitsAny isn't a PredicateError,
+			// and we don't need it here anyway.
+			podSchedulable.set(pod, schedulable, nil)
+			if schedulable {
+				glogx.V(4).UpTo(loggingQuota).Infof("Pod %s marked as unschedulable can be scheduled on %s. Ignoring in scale up.", pod.Name, nodeName)
 			}
-			continue
 		}
-		if nodeName, err := predicateChecker.FitsAny(pod, nodeNameToNodeInfo); err == nil {
-			glogx.V(4).UpTo(loggingQuota).Infof("Pod %s marked as unschedulable can be scheduled on %s. Ignoring in scale up.", pod.Name, nodeName)
-			podSchedulable.set(pod, true)
-		} else {
+
+		if !schedulable {
```

Comment:
Why refactor this whole method just because of minor change of signature of cache? Personally I find old version much easier to follow - check for cached result, run predicates if its missing.

Reply:
I thought this is what the new version does more clearly ("check for cached result, run predicates if missing"), i.e. follows the general pattern of:
- try to get data from cache,
- if missing, get it the hard way and put it in cache,
- do common logic on the data we wanted in the first place (whether it came from cache or not.)

Reply:
I guess it's personal preference. I see your point, but the common logic is a single line so duplication is minimal. If it was more I would extract it to function instead of refactoring like you described. The old way felt like a decision tree cache/no cache, schedulable/non schedulable. Only one variable on each level to think about. Here the conditions are more complex and I have to think if each possible combination is handled, which was trivial in old version.